### DISCUSSION TITLE
Problem with RSSI filter for bluetooth classic

### DISCUSSION THREAD

**Discussion by DysinLarshee** [mKeRix/room-assistant#755] [Q&A]
Currently, I have a Pi3 in my room for testing. The Pi3 is set to use Bluetooth Classic mode. When I am in the room, it works quite well. However, when I leave the room, and re-enter it after the state changed to not_home, the RSSI value would take about 2 minutes to go from a high value(-11) and slowly decrease to the range that I used to get when I sit still in the room(0 ~ -5). I set the minRssi to be -5, so the problem with this will be when I enter the room, I have to wait 2~3 minutes for the present state to change. 
May I know if there is any way to optimize this?

**Pi3 config:**
global:
  integrations:
    - bluetoothClassic
bluetoothClassic:
  minRssi: -5
  addresses:
    - '70:1F:3C:3E:71:E8'

**Log when entering the room**
08/06/2021, 23:48:37 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:48:42 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:48:42 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:48:47 - debug - BluetoothService: Command failed: hcitool -i hci0 cc "70:1F:3C:3E:71:E8" && hcitool -i hci0 rssi "70:1F:3C:3E:71:E8"
Can't create connection: Input/output error
Get connection info failed: No such file or directory

08/06/2021, 23:48:47 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:48:52 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:48:52 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:48:57 - debug - BluetoothService: Command failed: hcitool -i hci0 cc "70:1F:3C:3E:71:E8" && hcitool -i hci0 rssi "70:1F:3C:3E:71:E8"
Can't create connection: Input/output error
Get connection info failed: No such file or directory

08/06/2021, 23:48:57 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:49:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:49:02 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:02 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:07 - debug - BluetoothService: Command failed: hcitool -i hci0 cc "70:1F:3C:3E:71:E8" && hcitool -i hci0 rssi "70:1F:3C:3E:71:E8"
Can't create connection: Input/output error
Get connection info failed: No such file or directory

08/06/2021, 23:49:07 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:12 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:12 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:15 - debug - BluetoothService: Command failed: hcitool -i hci0 cc "70:1F:3C:3E:71:E8" && hcitool -i hci0 rssi "70:1F:3C:3E:71:E8"
Can't create connection: Input/output error
Get connection info failed: No such file or directory

08/06/2021, 23:49:15 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:22 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:22 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:23 - debug - BluetoothService: Command failed: hcitool -i hci0 cc "70:1F:3C:3E:71:E8" && hcitool -i hci0 rssi "70:1F:3C:3E:71:E8"
Can't create connection: Input/output error
Get connection info failed: No such file or directory

08/06/2021, 23:49:23 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:49:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:49:32 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:32 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:32 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:32 - debug - BluetoothClassicService: Received RSSI of -8.2 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:49:42 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:42 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:42 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:42 - debug - BluetoothClassicService: Received RSSI of -9.5 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:49:52 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:49:52 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:49:54 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:49:54 - debug - BluetoothClassicService: Received RSSI of -11.1 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:50:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:50:02 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:02 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:02 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:02 - debug - BluetoothClassicService: Received RSSI of -11.7 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:12 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:12 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:13 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:13 - debug - BluetoothClassicService: Received RSSI of -11 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:22 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:22 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:23 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:23 - debug - BluetoothClassicService: Received RSSI of -10.1 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:50:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:50:32 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:32 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:34 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:34 - debug - BluetoothClassicService: Received RSSI of -10 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:42 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:42 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:42 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:42 - debug - BluetoothClassicService: Received RSSI of -9.1 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:50:52 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:50:52 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:50:53 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:50:53 - debug - BluetoothClassicService: Received RSSI of -8.5 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:51:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:51:02 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:02 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:03 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:03 - debug - BluetoothClassicService: Received RSSI of -7.5 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:12 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:12 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:13 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:13 - debug - BluetoothClassicService: Received RSSI of -6.9 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:22 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:22 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:23 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:23 - debug - BluetoothClassicService: Received RSSI of -6.5 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:51:30 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:51:32 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:32 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:32 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:32 - debug - BluetoothClassicService: Received RSSI of -6.3 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:42 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:42 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:42 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:42 - debug - BluetoothClassicService: Received RSSI of -5.6 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:51:52 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:51:52 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:51:53 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:51:53 - debug - BluetoothClassicService: Received RSSI of -5.6 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking instance as online
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking room-assistant-pi3-bluetooth-classic-inquiries-switch as online
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-size as online
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking room-assistant-pi3-status-cluster-leader as online
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker as online
08/06/2021, 23:52:00 - debug - HomeAssistantService: Marking room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8 as online
08/06/2021, 23:52:02 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:52:02 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:52:02 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:52:02 - debug - BluetoothClassicService: Received RSSI of -5.1 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:52:12 - debug - BluetoothService: Locking adapter 0
08/06/2021, 23:52:12 - debug - BluetoothService: Querying for RSSI of 70:1F:3C:3E:71:E8 using hcitool
08/06/2021, 23:52:13 - debug - BluetoothService: Unlocking adapter 0
08/06/2021, 23:52:13 - debug - BluetoothClassicService: Received RSSI of -4.7 for 70:1F:3C:3E:71:E8 from PI3
08/06/2021, 23:52:13 - debug - HomeAssistantService: Sending new state PI3 for room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8
08/06/2021, 23:52:13 - debug - HomeAssistantService: Sending new state home for room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8-tracker
08/06/2021, 23:52:13 - debug - HomeAssistantService: Sending new attributes {"distance":4.7,"lastUpdatedAt":"2021-06-08T15:52:13.702Z"} for room-assistant-bluetooth-classic-70-1f-3c-3e-71-e8

**Comment by mKeRix** [comment]
fyi, this is tracked in #775
